### PR TITLE
Add is_dilos for future ZTS updates

### PR DESCRIPTION
**Pull request by ikozhukhov**
Signed-off-by: Igor Kozhukhov <igor@dilos.org>

### Motivation and Context

Add is_dilos identification for future updates in ZTS for reduce merge conflicts

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [ ] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:

- [ ] My code follows the ZFS on Linux [code style requirements](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**contributing** document](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md).
- [ ] I have added [tests](https://github.com/zfsonlinux/zfs/tree/master/tests) to cover my changes.
- [ ] I have run the ZFS Test Suite with this change applied.
- [ ] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

### CODE REVIEW

**Comment by freqlabs** [comment]
Can't you use `is_illumos` on Dilos? I realize you may have to search and replace `is_dilos` for `is_illumos` in all the tests.

**Comment by ikozhukhov** [comment]
nope, because DilOS contain more Debian userland in primary tools PATH like coreutils, grep, sed, find, etc.
illumos has own tools in preferences.

**Comment by freqlabs** [comment]
Ah, so you can use `is_illumos` for parts that use tools that are the same between Dilos and other systems using the illumos kernel, and `is_dilos` for places where Dilos and other illumos differ.

**Comment by ikozhukhov** [comment]
yes, will fo that.
plan is: provide alternative platform with the same codebase OpenZFS and produce the same ZTS like buildbot.
right now i do merges to to my DilOS tree and use the same tools with ZoL : like `df`, `file` , etc, where i have to fix older illumos tools.
to have `is_dilos` can help me with it and reduce merge conflicts.

File: tests/zfs-tests/include/libtest.shlib [hunk 1]
```diff
@@ -119,6 +119,21 @@ function is_freebsd
...
+# Determine if this is a DilOS test system
+#
+# Return 1 if platform DilOS, 0 if otherwise
+
+function is_dilos
+{
+	typeset id=""
+	[[ -f /etc/os-release ]] && . /etc/os-release
+	if [[ $id == "dilos" ]]; then
+		return 1
```
Comment (freqlabs) [inline tests/zfs-tests/include/libtest.shlib]: This should be the opposite, return 0 if it is Dilos, 1 if not.

Comment (ikozhukhov) [reply tests/zfs-tests/include/libtest.shlib]: fixed, thanks

File: tests/zfs-tests/include/libtest.shlib [hunk 2]
```diff
@@ -119,6 +119,21 @@ function is_freebsd
 	fi
 }
 
+# Determine if this is a DilOS test system
+#
+# Return 1 if platform DilOS, 0 if otherwise
```
Comment (behlendorf) [inline tests/zfs-tests/include/libtest.shlib]: ~This logic is inverted from all of the other `is_` platform checks.  Can we keep the behavior consistent and return 0 when the platfirm is DilOS.~  Saw you just fixed this, but please update the comment too.

Comment (ikozhukhov) [reply tests/zfs-tests/include/libtest.shlib]: fixed

**Pull request merged**